src/mrinufft/operators/base.py

```diff
@@ -500,15 +500,15 @@ def smaps(self):
         return self._smaps
 
     @smaps.setter
-    def smaps(self, smaps):
+    def smaps(self, smaps, check_only=False):
         if smaps is None:
             self._smaps = None
         elif len(smaps) != self.n_coils:
             raise ValueError(
                 f"Number of sensitivity maps ({len(smaps)})"
                 f"should be equal to n_coils ({self.n_coils})"
             )
```

Comment:
I don't find any good reason to not also check that the remaining dimensions are matching self.shape as well. 

```suggestion
        elif smaps.shape != (self.n_coils, *self.shape):
            raise ValueError(
                f"smaps shape is {smaps.shape}, it should be"
                f"(n_coils, *shape): {(self.n_coils, *self.shape)}"
            )
```